test/util/ChartUtils/getDomainOfDataByKey.spec.tsx

```diff
@@ -0,0 +1,169 @@
+      const result = getDomainOfDataByKey(data, 'x', 'number');
+      expect(result).toEqual([-Infinity, Infinity]);
+    });
+
+    it('should understand scientific notation', () => {
+      const data = [{ x: 9 }, { x: 1e3 }, { x: 8e2 }];
+      const result = getDomainOfDataByKey(data, 'x', 'number');
+      expect(result).toEqual([9, 1000]);
+    });
+
+    it('when given positive and negative zeroes, should use the first one from the array as both min and max', () => {
+      const data = [{ x: -0 }, { x: 0 }];
+      const result = getDomainOfDataByKey(data, 'x', 'number');
+      expect(result).toEqual([-0, -0]);
+
+      const data2 = [{ x: 0 }, { x: -0 }];
+      const result2 = getDomainOfDataByKey(data2, 'x', 'number');
+      expect(result2).toEqual([0, 0]);
+    });
+
+    it('should flatten arrays of numbers', () => {
+      const data = [{ x: [4, 1] }, { x: [2, 16] }, { x: [8, 3] }];
+      const result = getDomainOfDataByKey(data, 'x', 'number');
+      expect(result).toEqual([1, 16]);
+    });
+
+    it('should treat stringified numbers as strings', () => {
+      const data = [{ x: '4' }, { x: '2' }, { x: '8' }, { x: '12' }];
+      const result = getDomainOfDataByKey(data, 'x', 'number');
+      /*
+       * Even though 2 is smaller number than 12, '12' is smaller string than '2'.
+       * This is 100% a bug but before we fix it let's talk about backwards compatibility
+       */
+      expect(result).toEqual(['12', '8']);
+    });
+
+    it('should ignore non-number values', () => {
+      const data = [{ x: 'twelve' }, { x: 4 }, { x: 8 }, { x: '0' }];
+      const result = getDomainOfDataByKey(data, 'x', 'number');
+      expect(result).toEqual([4, 8]);
+    });
+
+    it('should return [Infinity, -Infinity] if none of the data points is a number', () => {
+      const data = [{ x: 'yesterday' }, { x: 'today' }, { x: 'tomorrow' }];
+      const result = getDomainOfDataByKey(data, 'x', 'number');
+      expect(result).toEqual([Infinity, -Infinity]);
+    });
+
+    it('should return 0 if it is specified as a number', () => {
+      const data = [{ x: 0 }, { x: 100 }, { x: 200 }];
+      const result = getDomainOfDataByKey(data, 'x', 'number');
+      expect(result).toEqual([0, 200]);
+    });
+
+    it('should ignore filterNil parameter', () => {
+      const data = [{ x: 0 }, { x: 100 }, { x: 200 }];
+      const resultWithTrue = getDomainOfDataByKey(data, 'x', 'number', true);
+      const resultWithFalse = getDomainOfDataByKey(data, 'x', 'number', false);
+      expect(resultWithTrue).toEqual(resultWithFalse);
+    });
+
+    it('should ignore 0 if it is specified as a string', () => {
+      const data = [{ x: '0' }, { x: '100' }, { x: '200' }];
+      const result = getDomainOfDataByKey(data, 'x', 'number');
+      expect(result).toEqual(['100', '200']);
+    });
+
+    it('should use the same number as both min and max if there is only one data point', () => {
```

Comment:
Indeed @nikolasrieble here's a test for the single value case. I don't understand the usefulness of such domain but I'm still discovering what all this stuff is for.